### PR TITLE
Use same queue for batch ops and for shutdown in h2_ssl_cert_test

### PR DESCRIPTION
**Pull request by yashykt**
Better fix for #13477

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

File: test/core/end2end/h2_ssl_cert_test.cc [hunk 1]
```diff
@@ -272,25 +270,11 @@ static void drain_cq(grpc_completion_queue* cq) {
...
-  grpc_server_shutdown_and_notify(f->server, f->shutdown_cq, tag(1000));
-  GPR_ASSERT(grpc_completion_queue_pluck(f->shutdown_cq, tag(1000),
-                                         grpc_timeout_seconds_to_deadline(5),
-                                         nullptr)
-                 .type == GRPC_OP_COMPLETE);
+  grpc_server_shutdown_and_notify(f->server, f->cq, tag(1000));
+  grpc_event ev = grpc_completion_queue_next(
+      f->cq, grpc_timeout_seconds_to_deadline(5), nullptr);
+  ASSERT_EQ(ev.type, GRPC_OP_COMPLETE);
+  ASSERT_EQ(ev.tag, tag(1000));
```
Comment (sreecha) [inline test/core/end2end/h2_ssl_cert_test.cc]: While this should be ideally true, I am worried that if the `f->cq` has some other pending operation (i.e with a different tag), then `grpc_completion_queue_next` may return that tag instead.
(**Note**: This ideally shouldn't happen but see that in `end_test(...)` function, we actually call `drain_cq(f->cq)` which just keeps calling completion queue next in a loop to drain all events.)

I suggest doing the following instead:
Call  the following here:
``` C++
 grpc_server_shutdown_and_notify(f->server, f->cq, tag(1000));
 grpc_completion_queue_shutdown(f->cq);
 drain_cq(f->cq); // This should only return once all the events (including the server shutdown) are done
```

And delete the following lines from the function `end_test(...)`
```C++
  grpc_completion_queue_shutdown(f->cq);
  drain_cq(f->cq)
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: test/core/end2end/h2_ssl_cert_test.cc (hunk 1 above)
Comment (yashykt) [reply test/core/end2end/h2_ssl_cert_test.cc]: Done!

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: test/core/end2end/h2_ssl_cert_test.cc [hunk 2]
```diff
@@ -304,11 +288,7 @@ static void shutdown_client(grpc_end2end_test_fixture* f) {
 static void end_test(grpc_end2end_test_fixture* f) {
   shutdown_client(f);
   shutdown_server(f);
-
-  grpc_completion_queue_shutdown(f->cq);
-  drain_cq(f->cq);
   grpc_completion_queue_destroy(f->cq);
-  grpc_completion_queue_destroy(f->shutdown_cq);
```
Comment (sreecha) [inline test/core/end2end/h2_ssl_cert_test.cc]: Unfortunately `grpc_completion_queue_destroy(f->shutdown_cq)` must still be called though - else its a memory leak
(I know its a very weird contract - but the fixture expects all tests to call destroy on `f->shutdown_cq` during test-teardown)

Comment (yashykt) [reply test/core/end2end/h2_ssl_cert_test.cc]: so, even if it's not being used, it must be created and destroyed?

Comment (yashykt) [reply test/core/end2end/h2_ssl_cert_test.cc]: This is a standalone test btw. It's not mixed with other filters/tests. If I remove the create, then I don't need to destroy either.

Comment (sreecha) [reply test/core/end2end/h2_ssl_cert_test.cc]: yeah..if you removed the create you wouldn't need to destroy. I missed the "removing the create" part :)

**Pull request merged**